### PR TITLE
Create Table migrations and models

### LINKED ISSUE #1: Create Bookings Table & Model
ID
Yard ID
Renter ID
Status: enum [pending, approved, rejected]
Booking Name
Date
Time
Duration
Description

### LINKED ISSUE #2: Create Yards Table & Model
ID
Host ID (renamed from User reference)
Name
Address
City
State
Zipcode
Price
Description
Availability: "string"
Payment: "string"
url_1
url_2
url_3

### LINKED ISSUE #3: Create Yard Purposes Table & Model
- ID
- yard_id
- purpose_id

### LINKED ISSUE #4: Create Purposes Table & Model
- ID
- Name: string

### PR DESCRIPTION
**Pull request by jbranham1**
I would like for @amsmyth1 along with others outside our pair before merged please!

Added all table migrations for engine
Added specs for models
Added relationships and validations in each model
Added factories

I thought maybe we should validate presence of those IDs(host_id, renter_id) that are typically references but will not be in this case sense the user table is on the frontend!

Close #1 
Close #2  
Close #3 
Close #4

### CODE REVIEW

**Review by Gvieve — APPROVED** [review]
Looks great. Thanks for building all of this out so quickly!

File: spec/factories/bookings.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FactoryBot.define do
+  factory :booking do
+    renter_id { 1 }
+    status { :pending }
+    booking_name { "My Event" }
+    date { "2021-04-13" }
+    time { "2021-04-13 16:19:51" }
+    duration { 1 }
+    description { "MyString" }
+
```
Comment (amsmyth1) [inline spec/factories/bookings.rb:10]: You are a champ for setting this up!

**Review by amsmyth1 — APPROVED** [review]
This is perfect, great (fast!!) work Jenny!

**Comment by amsmyth1** [comment]
Good thinking validating the renter and host ids since they are not technically foreign keys!

**Pull request merged**